src/main/java/org/broadinstitute/hellbender/tools/walkers/longreads/RecoverUncorrectedReads.java

```diff
@@ -0,0 +1,116 @@
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMFileGATKReadWriter;
+import picard.cmdline.programgroups.ReadDataManipulationProgramGroup;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Recover uncorrected reads
+ *
+ * <h3>Input</h3>
+ * <ul>
+ *     <li>The uncorrected BAM file</li>
+ *     <li>The CCS-corrected BAM file</li>
+ * </ul>
+ *
+ * <h3>Outputs</h3>
+ * <ul>
+ *     <li>A BAM file of remaining subreads that were not corrected by the CCS step</li>
+ * </ul>
+ *
+ * <h3>Usage Example</h3>
+ * <h4>Quickly count errors</h4>
+ * <pre>
+ *   gatk RecoverUncorrectedReads \
+ *     -I input.bam \
+ *     -C ccs.bam \
+ *     -O output.bam
+ * </pre>
+ */
+@DocumentedFeature
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = "Recover uncorrected reads",
+        oneLineSummary = "Recover uncorrected reads",
+        programGroup = ReadDataManipulationProgramGroup.class
+)
+public final class RecoverUncorrectedReads extends ReadWalker {
+    @Argument(fullName = "corrected",
+            shortName = "C",
+            doc="CCS-corrected reads")
+    public String corrected;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="Write output to this file")
+    public String output;
+    private SAMFileGATKReadWriter outputWriter;
+
+    private final Set<String> correctedReadNames = new HashSet<>();
+    private int uncorrectedReads = 0;
+
+    @Override
+    public void onTraversalStart() {
+        final SamReaderFactory srf = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT);
+
+        try (final SamReader srs = srf.open(IOUtils.getPath(corrected))) {
+            for ( final SAMRecord sr : srs ) {
+                String[] name        = sr.getReadName().split("/");
+                String   readZmwName = name[ 0 ] + "/" + name[ 1 ];
+                correctedReadNames.add(readZmwName);
+            }
+
+        } catch (final IOException e) {
+            throw new UserException("Unable to open path: " + corrected, e);
+        }
+
+        outputWriter = createSAMWriter(new GATKPath(IOUtils.getPath(output).toUri().toString()), true);
+    }
+
+    @Override
+    public void apply( GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext ) {
+        String[] name = read.getName().split("/");
+        if (name.length >= 2) {
+            String readZmwName = name[0] + "/" + name[1];
+            if (!correctedReadNames.contains(readZmwName)) {
+                outputWriter.addRead(read);
+                uncorrectedReads++;
+
+                logger.debug("uncorrected: {}", readZmwName);
+            } else {
+                logger.debug("corrected: {}", readZmwName);
```

Comment:
these debug loggers are going to print a lot and slow the tool down. Is this necessary? at the very least replace these with lambda funcitons as the java format code that runs on these is painfully slow and neednt be called if we are not in print debug mode